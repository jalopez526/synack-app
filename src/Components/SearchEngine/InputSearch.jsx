import React from "react";
import Input from "../Input";

const InputSearch = ({ value, onChange }) => {
  return (
    <Input
      type="text"
      data-testid="search-input"
      className="control"
      placeholder="What would you like to search?"
      value={value}
      onChange={onChange}
    />
  );
};

export default InputSearch;
