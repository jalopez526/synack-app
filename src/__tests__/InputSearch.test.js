import { render, fireEvent } from "@testing-library/react";
import InputSearch from "../Components/SearchEngine/InputSearch";
import renderer from 'react-test-renderer';

describe("Testing InputSearch component", () => {
  const onChange = jest.fn();

  it('renders correctly', () => {
    const tree = renderer
      .create(<InputSearch onChange={onChange} value="http://www.facebook.com" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("render Select component and selecting option", () => {
    const { getByTestId } = render(<InputSearch onChange={onChange} />);
    const element = getByTestId("search-input");

    fireEvent.change(element, { target: { value: "Hello" } });
  
    expect(element).toBeInTheDocument();
    expect(onChange).toHaveBeenCalled();
    expect(element.value).toBe("Hello");
  });
});
