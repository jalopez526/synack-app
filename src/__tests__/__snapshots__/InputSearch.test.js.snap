// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing InputSearch component renders correctly 1`] = `
<input
  className="control"
  data-testid="search-input"
  onChange={[MockFunction]}
  placeholder="What would you like to search?"
  type="text"
  value="http://www.facebook.com"
/>
`;
